Clarifying usage of environment variables on the command line

angrbrd opened the pull request:
## Description

These changes are documentation changes only. They are put in place to address several concerns with confusion on the usage of environment variables on the command line.
## Motivation and Context

The change adds modifications to existing examples of command line usage for deploy, invoke, query, and starting up docker containers where environment variables are used.

Fixes #1458
## How Has This Been Tested?

Documentation changes only. No tests required.
## Checklist:
- [x] I have added a [Signed-off-by](https://github.com/hyperledger/fabric/blob/master/CONTRIBUTING.md#legal-stuff)
- [x] Either no new documentation is required by this change, OR I added new documentation
- [x] Either no new tests are required by this change, OR I added new tests
- [x] I have run [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports), [go vet](https://golang.org/cmd/vet/), and [golint](https://github.com/golang/lint). I have cleaned up all valid errors and warnings in code I have added or modified. These tools may generate false positives. Don't be worried about ignoring some errors or warnings. The goal is clean, consistent, and readable code.

Fixes #1458

Signed-off-by: Anna D Derbakova adderbak@us.ibm.com

Commit:
Clarifying the usage of environment variables on the cmd line

Signed-off-by: Anna D Derbakova <adderbak@us.ibm.com>

Commit:
Clarifying usage of environment variables with docker containers

Signed-off-by: Anna D Derbakova <adderbak@us.ibm.com>

Commit:
Clarifying the process of logging in a user with docker containers

Signed-off-by: Anna D Derbakova <adderbak@us.ibm.com>

srderson commented on docs/API/SandboxSetup.md:
```diff
@@ -30,7 +30,7 @@ To set up the local development environment with security enabled, you must firs
 
 ###Vagrant Terminal 1 (validating peer)
-**Note:** To run with security enabled, first modify the [core.yaml](https://github.com/hyperledger/fabric/blob/master/peer/core.yaml) configuration file to set the <b>security.enabled</b> value to 'true' before building the peer executable. Alternatively, you can enable security by running the peer with environment variable CORE_SECURITY_ENABLED=true. To enable privacy and confidentiality of tra ...
+**Note:** To run with security enabled, first modify the [core.yaml](https://github.com/hyperledger/fabric/blob/master/peer/core.yaml) configuration file to set the <b>security.enabled</b> value to 'true' before building the peer executable. Alternatively, you can enable security by running the peer with environment variable CORE_SECURITY_ENABLED=true. To enable privacy and confidentiality of tra ...
```
Could you use ` around
`security.enabled`
`CORE_SECURITY_ENABLED=true`
`true`

srderson commented on docs/API/SandboxSetup.md:
````diff
@@ -152,9 +152,9 @@ POST host:port/chaincode
 ```
 
-**Note:** When security is enabled, modify the CLI command and the REST API payload to pass the <b>enrollmentID</b> of a logged in user. To log in a registered user through the CLI or the REST API, follow the instructions in the [note on security functionality](#note-on-security-functionality). On the CLI, the <b>enrollmentID</b> is passed with the <b>-u</b> parameter; in the REST API, the <b>enr ...
+**Note:** When security is enabled, modify the CLI command and the REST API payload to pass the <b>enrollmentID</b> of a logged in user. To log in a registered user through the CLI or the REST API, follow the instructions in the [note on security functionality](#note-on-security-functionality). On the CLI, the <b>enrollmentID</b> is passed with the <b>-u</b> parameter; in the REST API, the <b>enr ...
````
Same comment. I think ` may be a better choice than <b>

srderson commented on docs/API/SandboxSetup.md:
````diff
@@ -224,9 +224,9 @@ POST host:port/chaincode
 ```
 
-**Note:** When security is enabled, modify the CLI command and REST API payload to pass the <b>enrollmentID</b> of a logged in user. To log in a registered user through the CLI or the REST API, follow the instructions in the [note on security functionality](#note-on-security-functionality). On the CLI, the <b>enrollmentID</b> is passed with the <b>-u</b> parameter; in the REST API, the <b>enrollm ...
+**Note:** When security is enabled, modify the CLI command and REST API payload to pass the <b>enrollmentID</b> of a logged in user. To log in a registered user through the CLI or the REST API, follow the instructions in the [note on security functionality](#note-on-security-functionality). On the CLI, the <b>enrollmentID</b> is passed with the <b>-u</b> parameter; in the REST API, the <b>enrollm ...
````
Same comment. Realize these were <b> before your change. Maybe it can be cleaned up in another PR, but would like to get your opinion.

angrbrd replied:
I have gone through both of the files in this PR to make sure that all instances of <b>xxx</b> have been replaced with `...` where appropriate. About to push new changes.

srderson commented on docs/dev-setup/devnet-setup.md:
```diff
@@ -50,29 +50,40 @@ By default, we are using a consensus plugin called `NOOPS`, which doesn't really
+You can start up a few more validating peers in the similar manner as you wish. Remember to change the peer ID and add the enrollID/enrollSecret to the membersrvc.yaml.
+
+### Enroll/Login a test user (if security is enabled):
+If security is enabled, you must enroll a user with the certificate authority before sending requests. Choose a user that is already registered, i.e. added to the membersrvc.yaml file. Then, execute the command below to log in the user on the target validating peer. `CORE_PEER_ADDRESS` specifies the target validating peer for which the user is to be logged in.
```
Consider making "membersrvc.yaml" a link? [membersrvc.yaml](https://github.com/hyperledger/fabric/blob/master/membersrvc/membersrvc.yaml)

angrbrd replied:
Done.

srderson commented on docs/dev-setup/devnet-setup.md:
````diff
@@ -50,29 +50,40 @@ By default, we are using a consensus plugin called `NOOPS`, which doesn't really
+CORE_PEER_ADDRESS=172.17.0.2:30303 ./peer network login jim
+```
+
+**Note:**, the certificate authority allows the enrollID and enrollSecret credentials to be used only *once*. Therefore, login by the same user from any other validating peer will result in an error. Currently, the application layer is responsible for duplicating the crypto material returned from the CA to other peer nodes. If you want to test secure transactions from more than one peer node with ...
````
Capitalize 't' in 'the' to be consistent with other Notes

angrbrd replied:
Done.

srderson commented:
Great, nice to see cleanup in docs!

Commit:
Modifying all instances of <b> to `

Making formatting edits like replacing all instances of <b>xxx</b> to
`xxx` and adding required membersrvc.yaml links.

Signed-off-by: Anna D Derbakova <adderbak@us.ibm.com>

Commit:
Minor edits and adding required links

Making minor formatting edits and adding required membersrvc.yaml links.

Signed-off-by: Anna D Derbakova <adderbak@us.ibm.com>

angrbrd commented:
Didn't realize I'm in conflict now... let me rebase.
